src/loaders/OMETiffLoader.js

```diff
@@ -324,87 +116,72 @@ export default class OMETiffLoader {
     };
   }
 
-  async _getChannel({ image, x, y, z, signal }) {
-    const { tileSize, pool, isInterleaved: interleave } = this;
-    const { height, width } = this._getTileExtent({
-      x,
-      y,
-      z
-    });
-    // Passing in the height and width explicitly prevents resampling that geotiff does without such parameters.
-    const rasters = await image.readRasters({
-      window: [
-        x * tileSize,
-        y * tileSize,
-        x * tileSize + width,
-        y * tileSize + height
-      ],
-      pool,
-      signal,
-      width,
-      height,
-      interleave
-    });
-    const data = interleave ? rasters : rasters[0];
-    if (signal?.aborted) return null;
-    if (this.omexml.Type.startsWith('int')) {
-      // Uint view isn't correct for underling buffer, need to take an
-      // IntXArray view and cast to UintXArray.
-      if (data.BYTES_PER_ELEMENT === 1) {
-        return new Uint8Array(new Int8Array(data.buffer));
+  _getIFDIndex({ z = 0, channel, time = 0 }, zIndex = 0) {
+    let channelIndex;
+    // Without names, enforce a numeric channel indexing scheme
+    const channelNames = this.dimensions.find(dim => dim.field === 'channel')
+      .values;
+    if (channelNames.every(v => !v)) {
+      console.warn(
+        'No channel names found in OMEXML.  Please be sure to use numeric indexing.'
+      );
+      channelIndex = channel;
+    } else if (typeof channel === 'string') {
+      channelIndex = channelNames.indexOf(channel);
+    } else if (typeof channel === 'number') {
+      channelIndex = channel;
+    } else {
+      throw new Error('Channel selection must be numeric index or string');
+    }
+    const sizeZ = this.dimensions.find(dim => dim.field === 'z').values.length;
+    const sizeT = this.dimensions.find(dim => dim.field === 'time').values
+      .length;
+    const sizeC = this.dimensions.find(dim => dim.field === 'channel').values
+      .length;
+    const dimensionOrder = this.dimensions
+      .map(dim => dim.field[0])
+      .reverse()
+      .join('');
```

Comment:
Would not be opposed to specifying this convention in the docs explicitly - the whole "dimensions" concept is really strong and should probably have a schema, docs etc.

Reply:
My feeling is that `dimensions` is perhaps _too_ strong of a concept and I'd like to re-evaluate the schema considering how we use it in practice. My opinion is that it is overly complicated, and we only use a minor subset of the available fields.